source/usr/lib/mcvirt/parser.py

```diff
@@ -1201,3 +1224,16 @@ def parse_arguments(self, script_args=None):
                 user = user_factory.get_user_by_username(args.remove_username)
                 rpc.annotate_object(user)
                 user.delete()
+        elif action == 'vnc':
+            if args.vnc_action == 'connect':
+                vm_factory = rpc.get_connection('virtual_machine_factory')
```

Comment:
The VM factory, vm_object and annotation can be created in the previous scope (in the elif action == 'vnc')